distribution/tools/geoip-cli/src/main/java/org/elasticsearch/geoip/GeoIpCli.java

```diff
@@ -0,0 +1,161 @@
+
+import joptsimple.OptionSet;
+import joptsimple.OptionSpec;
+import org.elasticsearch.cli.Command;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.common.SuppressForbidden;
+import org.elasticsearch.common.hash.MessageDigests;
+import org.elasticsearch.common.io.PathUtils;
+import org.elasticsearch.common.xcontent.XContentGenerator;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+
+public class GeoIpCli extends Command {
+
+    private static final byte[] EMPTY_BUF = new byte[512];
+
+    // visible for testing
+    final OptionSpec<String> sourceDirectory;
+    final OptionSpec<String> targetDirectory;
+
+    public GeoIpCli() {
+        super("A CLI tool to prepare local GeoIp database service", () -> {});
+        sourceDirectory = parser.acceptsAll(Arrays.asList("s", "source"), "Source directory").withRequiredArg().required();
+        targetDirectory = parser.acceptsAll(Arrays.asList("t", "target"), "Target directory").withRequiredArg();
+
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options) throws Exception {
+        Path source = getPath(options.valueOf(sourceDirectory));
+        String targetString = options.valueOf(targetDirectory);
+        Path target = targetString != null ? getPath(targetString) : source;
+        copyTgzToTarget(terminal, source, target);
+        packDatabasesToTgz(terminal, source, target);
+        createOverviewJson(terminal, target);
+    }
+
+    @SuppressForbidden(reason = "file arg for cli")
+    private Path getPath(String file) {
+        return PathUtils.get(file);
+    }
+
+    private void copyTgzToTarget(Terminal terminal, Path source, Path target) throws IOException {
+        if (source.equals(target)) {
+            return;
+        }
+        List<Path> toCopy = Files.list(source).filter(p -> p.getFileName().toString().endsWith(".tgz")).collect(Collectors.toList());
+        for (Path path : toCopy) {
+            Files.copy(path, target.resolve(path.getFileName()), StandardCopyOption.REPLACE_EXISTING);
+        }
+    }
+
+    private void packDatabasesToTgz(Terminal terminal, Path source, Path target) throws IOException {
+        List<Path> toPack = Files.list(source).filter(p -> p.getFileName().toString().endsWith(".mmdb")).collect(Collectors.toList());
```

Comment:
Should we also pack txt files? There could be license and notice files?

Reply:
I don't think we have to, if users downloaded database from MaxMind they already had a chance to check LICENSE.txt, COPYRIGHTS.txt and so on, no need to copy it over to the tgz and ingest nodes. WDYT?

Reply:
That is true.